.github/workflows/pythonapp.yml

```diff
@@ -4,16 +4,22 @@ on: [push]
 
 jobs:
   build:
-
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up Python 3.7
-      uses: actions/setup-python@v1
-      with:
-        python-version: 3.7
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install -r requirements.txt
+      - uses: actions/checkout@v2
+      - name: Set up Python 3.7
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.7
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+      - name: Run flake8 linter
+        run: |
+          flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+          flake8 . --count --exit-zero --max-complexity=10 --max-line-length=120 --statistics
+      - name: Run black code formatter
+        run: |
+          black --check ./
```

Comment:
I would manually install flake8 and black here as they are not actual requirements (They don't need to be part of the docker image for it to work).
An alternative would be to have a different requirements file:
`requirements-tst.txt` with test utils (pytest-report, mypy, black, flake8...)

Reply:
I added it to the `requirements.txt` to make it available to everyone at this moment. What would be better, create the `requirements-tst.txt` or install each dependency before running it?

For instance:
```yml
- name: Run flake8 linter
    run: |
        pip install flake8 
        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
        flake8 . --count --exit-zero --max-complexity=10 --max-line-length=120 --statistics

Reply:
If you use poetry you can add packages as development packages and others for production. It's cleaner.

Reply:
@jbagot, I never worked with poetry. Right now, while we don't migrate to poetry, how would be the best way to have `black` and `flake8` to be installed, on requirements.txt or on the github actions file?